sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClient.java

```diff
@@ -0,0 +1,650 @@
+import com.azure.containers.containerregistry.models.RegistryArtifactProperties;
+import com.azure.containers.containerregistry.models.RepositoryProperties;
+import com.azure.containers.containerregistry.models.TagProperties;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.exception.ClientAuthenticationException;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.PagedResponse;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.SerializerAdapter;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/** Initializes a new instance of the asynchronous ContainerRegistry type. */
+@ServiceClient(builder = ContainerRegistryClientBuilder.class, isAsync = true)
+public final class ContainerRepositoryAsyncClient {
+    private final ContainerRegistryRepositoriesImpl serviceClient;
+    private final ContainerRegistriesImpl registriesImplClient;
+    private final String repositoryName;
+    private final String endpoint;
+    private final String apiVersion;
+
+    private final ClientLogger logger = new ClientLogger(ContainerRepositoryAsyncClient.class);
+
+    ContainerRepositoryAsyncClient(String repositoryName, HttpPipeline httpPipeline, SerializerAdapter serializerAdapter, String endpoint, String apiVersion) {
+        if (repositoryName == null) {
+            throw logger.logExceptionAsError(new NullPointerException("'repositoryName' can't be null"));
+        }
+
+        ContainerRegistryImpl registryImpl = new ContainerRegistryImplBuilder()
+            .pipeline(httpPipeline)
+            .serializerAdapter(serializerAdapter)
+            .url(endpoint).buildClient();
+
+        this.endpoint = endpoint;
+        this.repositoryName = repositoryName;
+        this.registriesImplClient = registryImpl.getContainerRegistries();
+        this.serviceClient = registryImpl.getContainerRegistryRepositories();
+        this.apiVersion = apiVersion;
+    }
+
+    /**
+     * Get endpoint associated with the class.
+     * @return String the endpoint associated with this client.
+     */
+    public String getEndpoint() {
+        return this.endpoint;
+    }
+
+
+    /**
+     * Get the registry associated with the client.
+     * @return Return the registry name.
+     */
+    public String getRegistry() {
+        return this.endpoint;
+    }
+
+    /**
+     * Get repository associated with the class.
+     * @return Return the repository name.
+     * */
+    public String getRepository() {
```

Comment:
Should this be called `getRepositoryName`?

Reply:
I intially did name it as that however went with the agreed upon apiview as I did not see any Java specific guideline for this unless I missed.

---
In reply to: [609151387](https://github.com/Azure/azure-sdk-for-java/pull/20368#discussion_r609151387) [](ancestors = 609151387)